ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/CommandSubmitterITSpec.scala

```diff
@@ -0,0 +1,259 @@
+    )
+  )
+}
+
+class CommandSubmitterITSpec
+    extends AsyncFlatSpec
+    with SandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues {
+
+  it should "populate participant with create, consuming and nonconsuming exercises" in {
+
+    val foo1Config = WorkflowConfig.SubmissionConfig.ContractDescription(
+      template = "Foo1",
+      weight = 1,
+      payloadSizeBytes = 100,
+    )
+    val foo2Config = WorkflowConfig.SubmissionConfig.ContractDescription(
+      template = "Foo2",
+      weight = 1,
+      payloadSizeBytes = 100,
+    )
+    val consumingExercisesConfig = ConsumingExercises(
+      probability = 1.0,
+      payloadSizeBytes = 100,
+    )
+    val nonconsumingExercisesConfig = NonconsumingExercises(
+      probability = 2.0,
+      payloadSizeBytes = 100,
+    )
+    val config = WorkflowConfig.SubmissionConfig(
+      numberOfInstances = 10,
+      numberOfObservers = 1,
+      uniqueParties = false,
+      instanceDistribution = List(
+        foo1Config,
+        foo2Config,
+      ),
+      nonconsumingExercises = Some(nonconsumingExercisesConfig),
+      consumingExercises = Some(consumingExercisesConfig),
+    )
+
+    for {
+      ledgerApiServicesF <- LedgerApiServices.forChannel(
+        channel = channel,
+        authorizationHelper = None,
+      )
+      apiServices = ledgerApiServicesF("someUser")
+      tested = CommandSubmitter(
+        names = new Names(),
+        benchtoolUserServices = apiServices,
+        adminServices = apiServices,
+        metricRegistry = new MetricRegistry,
+        metricsManager = NoOpMetricsManager(),
+      )
+      (signatory, observers) <- tested.prepare(config)
+      _ <- tested.submit(
+        config = config,
+        signatory = signatory,
+        observers = observers,
+        maxInFlightCommands = 1,
+        submissionBatchSize = 5,
+      )
+      eventsObserver = EventsObserver(expectedTemplateNames = Set("Foo1", "Foo2"))
+      _ <- apiServices.transactionService.transactionTrees(
+        config = WorkflowConfig.StreamConfig.TransactionTreesStreamConfig(
+          name = "dummy-name",
+          filters = List(
+            WorkflowConfig.StreamConfig.PartyFilter(
+              party = signatory.toString,
+              templates = List.empty,
+            )
+          ),
+          beginOffset = None,
+          endOffset = Some(LedgerOffset().withBoundary(LedgerOffset.LedgerBoundary.LEDGER_END)),
+          objectives = None,
+        ),
+        observer = eventsObserver,
+      )
+      observerResult: ObservedEvents <- eventsObserver.result
+    } yield {
+      observerResult.createEvents.size shouldBe config.numberOfInstances withClue ("number of create events")
+
+      observerResult.avgSizeOfCreateEventPerTemplateName("Foo1") shouldBe roughly(
+        foo1Config.payloadSizeBytes * 2,
+        toleranceMul = 0.5,
+      ) withClue ("payload size of create Foo1")
```

Comment:
```suggestion
      observerResult
        .avgSizeOfCreateEventPerTemplateName("Foo1")
        .toDouble shouldBe (foo1Config.payloadSizeBytes * 2.0 +- foo1Config.payloadSizeBytes.toDouble) withClue ("payload size of create Foo1")
```

Reply:
Is a custom `TripleEqualsSupport.Spread` needed? Wouldn't something like above do the same thing?

Reply:
Yes, I started with a code like that, but then wanted an easy way to say that we want this value with +- that many percents. 
It also has a nice property that it's enough to access `foo1Config.payloadSizeBytes` only once